test/quantization/test_quantize_fx.py

```diff
@@ -406,6 +406,122 @@ def forward(self, x):
+                return float_module
+
+        class QuantizedCustomModule(torch.nn.Module):
+            def __init__(self, conv):
+                super().__init__()
+                self.conv = conv
+
+            def forward(self, x):
+                return self.conv(x)
+
+            @classmethod
+            def from_observed(cls, observed_module):
+                assert hasattr(observed_module, 'qconfig')
+                assert hasattr(observed_module, 'activation_post_process')
+                observed_module.conv.activation_post_process = \
+                    observed_module.activation_post_process
+                quantized = cls(nnq.Conv2d.from_float(observed_module.conv))
+                return quantized
+
+        from torch.quantization import register_observed_custom_module_mapping
+        from torch.quantization import register_quantized_custom_module_mapping
+        register_observed_custom_module_mapping(CustomModule, ObservedCustomModule)
+        register_quantized_custom_module_mapping(CustomModule, QuantizedCustomModule)
+
+        class M(torch.nn.Module):
+            def __init__(self):
+                super().__init__()
+                self.conv = torch.nn.Conv2d(1, 1, 1)
+                self.custom = CustomModule()
+
+            def forward(self, x):
+                x = self.conv(x)
+                x = self.custom(x)
+                return x
+
+        class RefM(torch.nn.Module):
+            def __init__(self):
+                super().__init__()
+                self.conv1 = torch.nn.Conv2d(1, 1, 1)
+                self.conv2 = torch.nn.Conv2d(1, 1, 1)
+
+            def forward(self, x):
+                x = self.conv1(x)
+                x = self.conv2(x)
+                return x
+
+        data = torch.randn(1, 1, 1, 1)
+        # instantiate M and RefM and align the parameters
+        original_m = M()
+        original_ref_m = RefM()
+        original_ref_m.conv1.weight = torch.nn.Parameter(original_m.conv.weight.detach())
+        original_ref_m.conv1.bias = torch.nn.Parameter(original_m.conv.bias.detach())
+        original_ref_m.conv2.weight = torch.nn.Parameter(original_m.custom.conv.weight.detach())
+        original_ref_m.conv2.bias = torch.nn.Parameter(original_m.custom.conv.bias.detach())
+
+        from torch.fx.symbolic_trace import Tracer
+
+        # define a custom tracer to not trace through the custom module
+
+        class CustomTracer(Tracer):
+            def is_leaf_module(self, m):
+                return (m.__module__.startswith('torch.nn') and
+                        not isinstance(m, torch.nn.Sequential)) or \
+                    isinstance(m, CustomModule)
+
+        m = CustomTracer().trace(original_m).eval()
+        qconfig_dict = {'': default_qconfig}
+        # check prepared model
+        m = prepare_static_fx(m, qconfig_dict)
+        # calibration
+        m(data)
+        # all activation observers are inserted in the top level module
```

Comment:
What is the expected behavior of prepare for custom modules? I am assuming it is a module swap, ie replace with ObservedCustomModule. However, it looks like we are assuming that activation observers are inserted by prepare. Wouldnt it be better to treat custom modules as black boxes? i.e the user specifies where the observers for weights and activations are inserted, the framework just replaces modules.

Reply:
we only insert the activation_post_process for the output, we are treating the module as a blackbox

Reply:
Why not let the user define everything?  If the module has more than one output like LSTMs, what is the expected behavior?

Reply:
we can extend the current code to support multiple outputs and allow user to specify whether each output is quantized or not.
the reason is that we need to know whether these outputs are quantized or not in order to correctly quantize the code following custom module.

currently we assume custom module has one input and one output and both of them are quantized.

Reply:
Can you add a test for LSTMs and ensure that we can handle multiple inputs and outputs, it will likely be the first case for custom modules.

Reply:
yeah i think so, I can add this in a separate PR